src/test/kotlin/org/example/model/TodoTest.kt

```diff
@@ -0,0 +1,128 @@
+package org.example.model
+
+import junit.framework.TestCase.assertEquals
+import junit.framework.TestCase.assertNotNull
+import org.junit.Assert.assertNotEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.assertThrows
+import java.time.LocalDate
+
+class TodoTest {
+    companion object {
+        const val DEFAULT_TITLE = "장보기"
+        private lateinit var defaultTodo: Todo
+    }
+
+    @Before
+    fun setUp() {
+        defaultTodo = Todo(title = DEFAULT_TITLE)
+    }
+
+    @Test
+    @DisplayName("제목이 빈 값이 아니면 todo가 정상적으로 생성된다")
+    fun creatTodo() {
+        assertNotNull(defaultTodo.title)
+        assertEquals(DEFAULT_TITLE, defaultTodo.title)
+
+        val otherTodo = Todo(title = "$DEFAULT_TITLE 를 하지 않기")
+        assertNotEquals(defaultTodo.id, otherTodo.id)
+    }
+
+    @Test
+    @DisplayName("todo가 제목을 가지지 않으면 예외가 발생한다")
+    fun mustHaveTitle() {
+        assertThrows<IllegalArgumentException> {
+            Todo(title = "")
+        }
+
+        assertThrows<IllegalArgumentException> {
+            Todo(title = "   ")
+        }
+    }
+
+    @Test
+    @DisplayName("todo가 생성될 때 생성 날짜가 자동으로 생성 당시의 날짜로 부여된다")
+    fun mustSetCreationTimeAutomatically() {
+        val creationTodoDate: LocalDate? = LocalDate.now()
+
+        assertNotNull(defaultTodo.createdDate)
+        assertEquals(creationTodoDate, defaultTodo.createdDate)
+    }
+
+    @Test
+    @DisplayName("todo가 생성될 때 id가 자동으로 부여된다")
+    fun mustSetIdAutomatically() {
+        assertNotNull(defaultTodo.id)
+    }
+
+    @Test
+    @DisplayName("todo 생성 시 완료 여부는 false로 부여된다")
+    fun initDoneIsTrue() {
+        val doneWhenCreated = defaultTodo.done
+
+        assertEquals(doneWhenCreated, false)
+    }
+
+    @Test
+    @DisplayName("완료 여부를 변경할 수 있다")
+    fun canUpdateDone() {
+        val initDone = defaultTodo.done
+
+        defaultTodo.updateDone()
+        val updatedDone = defaultTodo.done
+
+        assertEquals(initDone, !updatedDone)
+    }
+
+    @Test
+    @DisplayName("'완료' 상태일 때 실행하면 '미완료'로 변경되고 '미완료' 상태일 때 실행하면 '완료'로 변경된다")
+    fun toggleDone() {
+        val doneWhenCreated = defaultTodo.done
+
+        defaultTodo.updateDone()
+        val doneWhenCompleted = defaultTodo.done
+
+        defaultTodo.updateDone()
+        val doneWhenUncompleted = defaultTodo.done
+
+        assertEquals(doneWhenCreated, false)
+        assertEquals(doneWhenCompleted, !doneWhenCreated)
+        assertEquals(doneWhenUncompleted, !doneWhenCompleted)
```

Comment:
```suggestion
    fun toggleDone() {
        val doneWhenCreated = defaultTodo.done
        assertTrue(doneWhenCreated)

        defaultTodo.updateDone()
        assertFalse(defaultTodo.done)

        defaultTodo.updateDone()
        assertTrue(defaultTodo.done)
```

이런 흐름을 표현하는 테스트는 위처럼 작성하기도 하죠
왜일까요? 2가지만 생각해보세요!

Reply:
아래 2가지 이유 생각해봤습니다! 수정해주신 코드가 훨씬 좋네요 👍🏼
1. 원래 코드는 여러 변수를 마지막 한 번에 다 비교하는 방식이었는데 변경된 코드는 상태 변화 직후 즉시 검증하므로 가독성과 코드 이해 측면에서 더 좋다고 느껴집니다
2. 상태 값을 변경할 때마다 변수에 저장했지만 새 코드는 그럴 필요가 없습니다